### DISCUSSION TITLE
请教个关于：Lua table 转 JSON 的问题

### DISCUSSION THREAD

**Discussion by aceyin** [cloudwu/skynet#1878] [General]
使用 Mongo 的时候，必须将 `lua table` 转化为 JSON/BSON 之后才能保存到 Mongodb 里面。  

但如果 `lua table` 里面有 `不连续的数字 Key` ，那么在转化为 JSON/BSON 的时候，是会报错的。
> 比如道具表，里面的数据通常是 {[道具ID]=数量} 这种格式，如 {[501]=100, [280]=3}

针对这种问题，大家是怎么处理的？

**Comment by sniper00** [comment]
https://github.com/sniper00/moon/issues/163

**Comment by sniper00** [comment]
https://github.com/cloudwu/skynet/issues/1484

**Comment by fanyh** [comment]
> 使用 Mongo 的时候，必须将 `lua table` 转化为 JSON/BSON 之后才能保存到 Mongodb 里面。
> 
> 但如果 `lua table` 里面有 `不连续的数字 Key` ，那么在转化为 JSON/BSON 的时候，是会报错的。
> 
> > 比如道具表，里面的数据通常是 {[道具ID]=数量} 这种格式，如 {[501]=100, [280]=3}
> 
> 针对这种问题，大家是怎么处理的？

deep copy数据，将数字key转换为string
